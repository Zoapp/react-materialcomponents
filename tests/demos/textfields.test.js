import React from "react";
import renderer from "react-test-renderer";
import TextFields from "@demos/components/textfields";

describe("demos/components/textfields", () => {
  it("renders correctly", () => {
    const component = renderer.create(<TextFields />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
