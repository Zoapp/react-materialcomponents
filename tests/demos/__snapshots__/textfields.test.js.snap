// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`demos/components/textfields renders correctly 1`] = `
<section>
  <h1>
    TextField examples
  </h1>
  <div>
    <div
      className="mdc-text-field mdc-text-field--upgraded"
    >
      <input
        className="mdc-text-field__input"
        defaultValue={undefined}
        id="1"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="text"
      />
      <label
        className="mdc-floating-label"
        focused="false"
        htmlFor="1"
        style={Object {}}
      >
        Demo1
      </label>
      <div
        className="mdc-text-field__bottom-line"
      />
    </div>
  </div>
  <div
    style={
      Object {
        "padding": "16px",
      }
    }
  >
    <div
      className="mdc-text-field mdc-text-field--upgraded"
    >
      <input
        className="mdc-text-field__input"
        defaultValue="Hello"
        id="2"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="text"
      />
      <label
        className="mdc-floating-label mdc-floating-label--float-above"
        focused="false"
        htmlFor="2"
        style={Object {}}
      >
        Demo2
      </label>
      <div
        className="mdc-text-field__bottom-line"
      />
    </div>
  </div>
  <div
    style={
      Object {
        "padding": "16px",
      }
    }
  >
    <div
      className="mdc-text-field mdc-text-field--upgraded mdc-text-field--disabled"
    >
      <input
        className="mdc-text-field__input"
        defaultValue="Hello"
        disabled="disabled"
        id="3"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="text"
      />
      <label
        className="mdc-floating-label mdc-floating-label--float-above"
        focused="false"
        htmlFor="3"
        style={Object {}}
      >
        Demo2
      </label>
      <div
        className="mdc-text-field__bottom-line"
      />
    </div>
  </div>
  <div
    style={
      Object {
        "padding": "16px",
      }
    }
  >
    <div
      className="mdc-text-field mdc-text-field--upgraded mdc-text-field--dense"
    >
      <input
        className="mdc-text-field__input"
        defaultValue={undefined}
        id="4"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="text"
      />
      <label
        className="mdc-floating-label"
        focused="false"
        htmlFor="4"
        style={Object {}}
      >
        Demo1
      </label>
      <div
        className="mdc-text-field__bottom-line"
      />
    </div>
  </div>
  <div
    style={
      Object {
        "padding": "16px",
      }
    }
  >
    <div
      className="mdc-text-field mdc-text-field--upgraded mdc-text-field--with-leading-icon"
    >
      <i
        className="mdc-text-field__icon material-icons"
        onClick={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex="0"
      >
        event
      </i>
      <input
        className="mdc-text-field__input"
        defaultValue="Hello"
        id="5"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="text"
      />
      <label
        className="mdc-floating-label mdc-floating-label--float-above"
        focused="false"
        htmlFor="5"
        style={Object {}}
      >
        Demo2
      </label>
      <div
        className="mdc-text-field__bottom-line"
      />
    </div>
  </div>
  <div
    style={
      Object {
        "padding": "16px",
      }
    }
  >
    <div
      className="mdc-text-field mdc-text-field--upgraded mdc-text-field--with-trailing-icon mdc-text-field--disabled"
    >
      <input
        className="mdc-text-field__input"
        defaultValue="Hello"
        disabled="disabled"
        id="6"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="text"
      />
      <label
        className="mdc-floating-label mdc-floating-label--float-above"
        focused="false"
        htmlFor="6"
        style={Object {}}
      >
        Demo2
      </label>
      <i
        className="mdc-text-field__icon material-icons"
        onClick={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex="0"
      >
        delete
      </i>
      <div
        className="mdc-text-field__bottom-line"
      />
    </div>
  </div>
  <div
    style={
      Object {
        "padding": "16px",
      }
    }
  >
    <div
      className="mdc-text-field mdc-text-field--upgraded mdc-text-field--with-trailing-icon mdc-text-field--textarea"
    >
      <textarea
        className="mdc-text-field__input"
        defaultValue="Hello"
        id="7"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="text"
      />
      <label
        className="mdc-floating-label mdc-floating-label--float-above"
        focused="false"
        htmlFor="7"
        style={Object {}}
      >
        Demo2
      </label>
      <i
        className="mdc-text-field__icon material-icons"
        onClick={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex="0"
      >
        delete
      </i>
      <div
        className="mdc-text-field__bottom-line"
      />
    </div>
  </div>
  <div
    style={
      Object {
        "padding": "16px",
      }
    }
  >
    <div
      className="mdc-text-field mdc-text-field--upgraded mdc-text-field--with-trailing-icon mdc-text-field--fullwidth mdc-text-field--textarea"
    >
      <textarea
        aria-label="Demo2"
        className="mdc-text-field__input"
        defaultValue="Hello"
        id="8"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Demo2"
        type="text"
      />
      <i
        className="mdc-text-field__icon material-icons"
        onClick={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex="0"
      >
        delete
      </i>
      <div
        className="mdc-text-field__bottom-line"
      />
    </div>
  </div>
  <div
    style={
      Object {
        "padding": "16px",
      }
    }
  >
    <div
      className="zrmc-text-field-wrapper"
    >
      <div
        className="mdc-text-field mdc-text-field--upgraded mdc-text-field--invalid"
      >
        <input
          aria-controls="9-helper-text"
          className="mdc-text-field__input"
          defaultValue={undefined}
          id="9"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="text"
        />
        <label
          className="mdc-floating-label"
          focused="false"
          htmlFor="9"
          style={Object {}}
        >
          Demo2
        </label>
        <div
          className="mdc-text-field__bottom-line"
        />
      </div>
      <p
        aria-hidden="true"
        className="mdc-text-field-helper-text"
      >
        Must be at least 8 characters
      </p>
    </div>
  </div>
  <div
    style={
      Object {
        "padding": "16px",
      }
    }
  >
    <div
      className="mdc-text-field mdc-text-field--upgraded mdc-text-field--outlined mdc-text-field--dense"
    >
      <input
        className="mdc-text-field__input"
        defaultValue={undefined}
        id="10"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="text"
      />
      <label
        className="mdc-floating-label"
        focused="false"
        htmlFor="10"
        style={Object {}}
      >
        Demo1
      </label>
      <div
        className="mdc-text-field__outline"
      >
        <svg>
          <path
            className="mdc-text-field__outline-path"
          />
        </svg>
      </div>
      <div
        className="mdc-text-field__idle-outline"
      />
    </div>
  </div>
  <div
    style={
      Object {
        "padding": "16px",
      }
    }
  >
    <div
      className="zrmc-text-field-wrapper"
    >
      <div
        className="mdc-text-field mdc-text-field--upgraded mdc-text-field--invalid mdc-text-field--box"
      >
        <input
          aria-controls="11-helper-text"
          className="mdc-text-field__input"
          defaultValue={undefined}
          id="11"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="text"
        />
        <label
          className="mdc-floating-label"
          focused="false"
          htmlFor="11"
          style={Object {}}
        >
          Demo2
        </label>
        <div
          className="mdc-text-field__bottom-line"
        />
      </div>
      <p
        aria-hidden="true"
        className="mdc-text-field-helper-text"
      >
        Must be at least 8 characters
      </p>
    </div>
  </div>
  <h2>
    Textfield Properties
  </h2>
  <div
    className="mdc-layout-grid"
  >
    <div
      className="mdc-layout-grid__inner"
    >
      <div
        className="mdc-layout-grid__cell"
      >
        Name
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        Default
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        Type
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        mdcElement
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        MDC_TEXTFIELD
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        string
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        label
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        null
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        string
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        id
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        null
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        string
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        type
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        text
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        string
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        disabled
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        false
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        bool
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        onChange
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        () =&gt; {}
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        function
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        dense
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        false
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        bool
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        helperText
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        null
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        string
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        htPersistent
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        false
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        bool
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        htValidationMsg
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        false
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        bool
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        fullwidth
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        false
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        bool
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        isTextarea
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        false
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        bool
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        isBoxed
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        false
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        bool
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        outlined
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        false
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        bool
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        leadingIcon
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        null
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        string
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        trailingIcon
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        null
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        string
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        onClickLI
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        null
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        function
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        onClickTI
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        null
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        function
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        noFloatingLabel
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        false
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        bool
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        defaultValue
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        undefined
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        string
      </div>
    </div>
  </div>
</section>
`;
